standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/metasummary/MetaSummarySchema.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.metastore.metasummary;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+
+public class MetaSummarySchema {
+
+  private final LinkedHashSet<String> fields = new LinkedHashSet<>();
+
+  public MetaSummarySchema addFields(List<String> fields) {
+    if (fields != null) {
+      addFields(fields.toArray(new String[0]));
```

Comment:
Should we directly convert List to LinkedHashSet?
`fields.addAll(new LinkedHashSet<String>(fields));`

Reply:
yes, will fix it